### PR TITLE
Fix comment status

### LINKED ISSUE #573: Comments icon colored with no new comments
It appears that #515 contains a bug: In production I just posted an answer to a user comment on one of my media, and now I have a yellow comment icon in the navbar indicating that there are new comments, while clicking on the icon shows there are none:

new_comments

There is no way to decolor the yellow icon (only waiting until someone posts a new commmment somewhere and the clearing all commments). This is actually something we wanted to avoid in #515 but for some reason, it does not work.

### PR DESCRIPTION
**Pull request by Splines**
Fixes #573.

<sub>Previous PR for this was #576, closed in favor of this one as this directly branches off the new "dev" branch.</sub>

### Other things to do
- [x] Delete old branch of #574
- [x] Create migration to fix inconsistencies in db introduced with #515. ⚠ Should be done in conjunction with this PR. -> see #587 
- [x] Model tests ran through locally ✅

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -549,14 +549,36 @@ def subscribed_commentable_media_with_comments
...
+  # For each medium, the following information is stored:
+  # - the medium itself
+  # - the thread of the medium
+  # - the latest comment by somebody else than the current user
+  # - the latest comment by any user (which might include the current user)
+  def subscribed_media_with_latest_comments_not_by_creator
+    media = subscribed_commentable_media_with_comments.map do |m|
+      thread = m.commontator_thread
+      comments = thread.comments
+      next unless comments
```
Comment (fosterfarrell9) [inline app/models/user.rb:568]: `comments` is an array, and empty arrays are truthy in Ruby. You might want to replace this by 

```ruby
next if comments.blank?
```

File: app/models/user.rb [hunk 2]
```diff
@@ -549,14 +549,36 @@ def subscribed_commentable_media_with_comments
...
+
+      comments_not_by_creator = comments.reject { |c| c.creator == self }
+      next unless comments_not_by_creator
```
Comment (fosterfarrell9) [inline app/models/user.rb:571]: Same here. You might want to replace this by 

```ruby
next if comments_not_by_creator.blank?
```

File: app/controllers/readers_controller.rb [hunk 3]
```diff
@@ -9,7 +9,7 @@ def update
     @reader = Reader.find_or_create_by(user: current_user,
                                        thread: @thread)
     @reader.touch
-    @anything_left = current_user.media_latest_comments.any? do |m|
+    @anything_left = current_user.subscribed_media_with_latest_comments_not_by_creator.any? do |m|
       (Reader.find_by(user: current_user, thread: m[:thread])
             &.updated_at || 1000.years.ago) < m[:latest_comment].created_at
     end
```
Comment (fosterfarrell9) [inline app/controllers/readers_controller.rb:15]: In some later PR, the code duplication in `main_controller.rb` and `readers_controller.rb` should be removed, e.g. by writing a method for the last two lines.

**Review by fosterfarrell9 — CHANGES_REQUESTED** [review]

**Review by fosterfarrell9 — APPROVED** [review]

**Comment by Splines** [comment]
thanks for the approval. I will merge once the respective migration is ready

**Pull request merged**